sklearn/utils/validation.py

```diff
@@ -811,6 +811,10 @@ def check_array(
 
     if sp.issparse(array):
         _ensure_no_complex_data(array)
+
+        if accept_sparse is False:
+            accept_sparse = ["csr", "csc"]
```

Comment:
We can not do this because it changes the behavior of `accept_sparse` in `check_array` and is backward incompatible, which is causing many [CI failures](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=42371&view=logs&j=c0f56aa6-924f-525c-3d4d-cdeecf409e66&t=f81f9209-4a58-571a-6d88-f139846a65e1). 

To fix this issue, we need to adjust `label_ranking_average_precision_score` itself to accept `y_true` as sparse, by setting `accept_sparse="csr"` here:

https://github.com/scikit-learn/scikit-learn/blob/735d3c635096b7e1f9706001b6c40fb1a5d4b486/sklearn/metrics/_ranking.py#L1073

From looking at the code, it maybe a little difficult to do since `type_of_target` does not really work with sparse targets. The quick fix is to adjust `label_ranking_average_precision_score` as follows:

```python
    y_true = check_array(y_true, ensure_2d=False, accept_sparse="csr")
    
    ...
    if not issparse(y_true):
        # Handle badly formatted array and the degenerate case with one label
        y_type = type_of_target(y_true, input_name="y_true")
        if y_type != "multilabel-indicator" and not (
            y_type == "binary" and y_true.ndim == 2
        ):
            raise ValueError("{0} format is not supported".format(y_type))

        y_true = csr_matrix(y_true)
```

Reply:
Thanks for the feedback. I have included this change by adding the `if not issparse(y_true)` line in a subsequent now. Let me know if I'm missing anything.